GCSViews/ConfigurationView/ConfigMotorTest.cs

```diff
@@ -178,12 +198,71 @@ private int get_motormax()
             return motormax;
         }
 
+        private bool set_frame_class_and_type(string class_param_name, string type_param_name)
+        {
+            if (!MainV2.comPort.MAV.param.ContainsKey(class_param_name) || !MainV2.comPort.MAV.param.ContainsKey(type_param_name))
+            {
+                return false;
+            }
+            var frame_class = (int)MainV2.comPort.MAV.param[class_param_name].Value;
+            var class_list = ParameterMetaDataRepository.GetParameterOptionsInt(class_param_name, MainV2.comPort.MAV.cs.firmware.ToString());
+            foreach (var item in class_list)
+            {
+                if (item.Key == Convert.ToInt32(frame_class))
+                {
+                    FrameClass.Text = "Class: " + item.Value;
+                    break;
+                }
+            }
+
+            var frame_type = (int)MainV2.comPort.MAV.param[type_param_name].Value;
+            var type_list = ParameterMetaDataRepository.GetParameterOptionsInt(type_param_name, MainV2.comPort.MAV.cs.firmware.ToString());
+            foreach (var item in type_list)
+            {
+                if (item.Key == Convert.ToInt32(frame_type))
+                {
+                    FrameType.Text = "Type: " + item.Value;
+                    break;
+                }
+            }
+
+            lookup_frame_layout(frame_class, frame_type);
+
+            return true;
+        }
+
+
+        private void lookup_frame_layout(int frame_class, int frame_type)
+        {
+            try
+            {
+                using (StreamReader r = new StreamReader("C:/motor_layout.json"))

```

Comment:
This is not the correct way to do this. I could not work out how to include the file with the build.

Reply:
you can use this a a reference
https://github.com/ArduPilot/MissionPlanner/blob/6f95f65707e15b8e412432cc33aee22aa4daa868/Utilities/Firmware.cs#L124

Reply:
https://github.com/ArduPilot/MissionPlanner/blob/5dac1d51464f282735bf8d5d4bf7769ffa09c429/MissionPlanner.csproj#L4935

Reply:
need to add it to missionplannerlib as well

Reply:
done, thanks.